### PR TITLE
Add Profession model

### PR DESCRIPTION
**Pull request by paulohfev**

### CODE REVIEW

File: app/models/profession.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# frozen_string_literal: true
+
+class Profession < ApplicationRecord
+  has_many :users, dependent: :nullify
```
Comment (kaiomagalhaes) [inline app/models/profession.rb:4]: why nullify?

File: db/migrate/20230925013358_add_profession_ref_to_users.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal: true
+
+class AddProfessionRefToUsers < ActiveRecord::Migration[7.0]
+  def change
+    add_reference :users, :profession, null: false, foreign_key: true
```
Comment (kaiomagalhaes) [inline db/migrate/20230925013358_add_profession_ref_to_users.rb:5]: it should be possible for an user to exist without a profession. The way it is here we wouldn't be able to create the user without the profession

File: spec/factories/professions.rb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal: true
+
+FactoryBot.define do
+  factory :profession do
+    name { 'MyString' }
```
Comment (kaiomagalhaes) [inline spec/factories/professions.rb:5]: use the FFaker gem to define random professions

File: spec/models/profession_spec.rb [hunk 4]
```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe Profession, type: :model do
```
Comment (kaiomagalhaes) [inline spec/models/profession_spec.rb:5]: you need to add the test for the user, to should belong to professions

**Review by kaiomagalhaes — CHANGES_REQUESTED** [review]

File: app/models/profession.rb (hunk 1 above)
Comment (paulohfev) [reply app/models/profession.rb:4]: @kaiomagalhaes this way, if a profession is deleted, the associated users won't be deleted along with it. Any user that had a profession_id would be set to nil

File: spec/models/user_spec.rb [hunk 5]
```diff
@@ -12,6 +12,10 @@
     it { is_expected.to validate_uniqueness_of(:google_id) }
   end
 
+  describe 'should validate the associations' do
```
Comment (kaiomagalhaes) [inline spec/models/user_spec.rb:15]: rename this to 'associations', mostly because it is a pattern we've been following

**Review by kaiomagalhaes — CHANGES_REQUESTED** [review]

**Review by kaiomagalhaes — APPROVED** [review]

**Pull request merged**